src/common.rs

```diff
@@ -16,6 +18,19 @@ pub fn find_seat_price(
     find_seat_price_for_protocol_after_49(stakes, max_number_of_seats, minimum_stake_ratio)
 }
 
+/// With statelessnet feature enabled, we need to take into account `num_chunk_only_producer_seats`.
+pub fn find_max_number_of_seats(protocol_config: &RpcProtocolConfigResponse) -> NumSeats {
+    let seats_before_statelessnet = protocol_config.num_block_producer_seats
+        + protocol_config
+            .avg_hidden_validator_seats_per_shard
+            .iter()
+            .sum::<u64>();
+    std::cmp::max(
+        seats_before_statelessnet,
+        protocol_config.num_chunk_only_producer_seats,
+    )
+}
+
 /// This implementation is ported from near-api-js:
 /// https://github.com/near/near-api-js/blob/bdbf839f54fbc399d7299da8cf9966bbdc426238/packages/utils/src/validators.ts#L24-L50
 fn find_seat_price_for_protocol_before_49(
```

Comment:
Also, I think we don't need to check protocol version for the current change because `protocol_config.num_chunk_only_producer_seats` will be zero for older protocol versions, so it will not affect the result.